kore/src/Kore/Step/Axiom/EvaluationStrategy.hs

```diff
@@ -101,8 +101,20 @@ simplificationEvaluation rule =
     BuiltinAndAxiomSimplifier $ \term condition -> do
         results' <- evaluateAxioms [rule] condition term
         let initial = Step.toConfigurationVariables (Pattern.fromTermLike term)
+            remainders' = Results.remainders results'
         Step.recoveryFunctionLikeResults initial results'
-        return $ Results.toAttemptedAxiom results'
+        let attemptedAxiom = Results.toAttemptedAxiom results'
+        case attemptedAxiom of
```

Comment:
I think you can factor this `case` out to make the code more readable.

Reply:
I agree, this should be something like `when (hasRemainder attemptedAxiom) $ warnSimplificationWithRemainder ...`